okio/src/main/java/okio/Buffer.java

```diff
@@ -1283,9 +1292,9 @@ Segment writableSegment(int minimumCapacity) {
       if (s == null) {
         // No segments to scan!
         return -1L;
-      } else if (size - fromIndex < fromIndex) {
+      } else if (toIndex - fromIndex < fromIndex) {
```

Comment:
This should still be size. Otherwise incorrect. This is the offset of the scan, not the desired offset of the scan. It’ll be broken for buffers with multiple segments.